lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,551 @@
+            'timeout_in_seconds',
+            'tolerated_number_of_failures']
+        for item in monitor_config_list:
+            if azure_profile.get('monitor_config', {}).get(item) != properties.get('monitor_config', {}).get(item):
+                if bool(properties.get('monitor_config', {}).get(item)):
+                    result['monitor_config'][item] = properties.get('monitor_config', {}).get(item)
+
+        return result
+
+    def traffic_manager_to_dict(self, traffic_manager, resource_group, name, location, properties):
+        '''
+        Converts a traffic manager profile to a dictionary
+
+        :param name: name of a traffic  manager
+        :return: traffic manage object
+        '''
+        if not bool(traffic_manager):  # Create a stub if there is no traffic manager
+            if bool(properties) and len(properties) > 0:
+                monitor_config = properties.get('monitor_config', [])
+                dns_config = properties.get('dns_config', [])
+                tags = properties.get('tags', [])
+            else:
+                name = None
+                resource_group = None
+                dns_config = {}
+                monitor_config = {}
+                tags = []
+                properties = dict(
+                    dns_config=dns_config,
+                    monitor_config=monitor_config,
+                    tags=tags)
+
+            return dict(
+                # id=traffic_manager.profile.id,
+                name=name,
+                resource_group=resource_group,
+                location=location,
+                properties=properties)
+
+        elif traffic_manager:  # Create a dictionary from the Azure traffic manager
+            properties = dict(tags=traffic_manager.tags)
+            return dict(
+                id=traffic_manager.id,
+                name=traffic_manager.name,
+                location=traffic_manager.location,
+                properties=properties)
+
+    def get_traffic_manager_profile(self, resource_group, name):
+        '''
+        Fetch a traffic manager profile.
+
+        :param name: name of a traffic  manager
+        :return: traffic manage object
+        '''
+        try:
+            profiles = self.trafficmanager_client.profiles.list_by_resource_group(resource_group)
+            #  Check the profiles and return the one that matches the name
+            for profile in profiles:
+                if profile.name == name:
+                    return self.trafficmanager_client.profiles.get(resource_group, name)
+            return None  # if there is no profile returns None
+        except CloudError as cloud_error:
+            self.fail(
+                "Error getting traffic manager profile with nanme: {0}. {1}".format(name, cloud_error))
+        except Exception as exc:
+            self.fail("Error retrieving traffic manager {0} - {1}".format(name, str(exc)))
+
+    def exist_traffic_manager_profile(self, resource_group, name):
```

Comment:
Duplicate with `get` method? This logic can be handled as `if get_traffic_manager_profile`